src/Xamarin.Android.Build.Tasks/Tasks/ProcessNativeLibraries.cs

```diff
@@ -90,12 +106,32 @@ public override bool RunTask ()
 					}
 				}
 
-				output.Add (library);
+				if (!IgnoreLibraryWhenLinkingRuntime (library)) {
+					output.Add (library);
+				}
 			}
 
 			OutputLibraries = output.ToArray ();
 
 			return !Log.HasLoggedErrors;
 		}
+
+		bool IgnoreLibraryWhenLinkingRuntime (ITaskItem libItem)
+		{
+			if (!NativeRuntimeLinking) {
+				return false;
+			}
+
+			// We ignore all the shared libraries coming from the runtime packages, as they are all linked into our runtime and
+			// need not be packaged.
+			string packageId = libItem.GetMetadata ("NuGetPackageId");
+			if (packageId.StartsWith ("Microsoft.NETCore.App.Runtime.Mono.android-", StringComparison.OrdinalIgnoreCase) ||
+			    packageId.StartsWith ("Microsoft.NETCore.App.Runtime.CoreCLR.android-", StringComparison.OrdinalIgnoreCase)) {
+				return true;
+			}
```

Comment:
Can this use the existing helper:

https://github.com/dotnet/android/blob/c4a62e53cbe868d7592a55eeac09ffa1012ea165/src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs#L425-L430

This would also work for any runtime, looks like.